frontend/src/metabase/components/DashboardSelector/DashboardSelector.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { t } from "ttag";
+import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+import DashboardPicker from "metabase/containers/DashboardPicker";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { useDashboardQuery } from "metabase/common/hooks";
+import {
+  DashboardPickerContainer,
+  DashboardPickerButton,
+} from "./DashboardSelector.styled";
+
+interface DashboardSelectorProps {
+  onChange: (value?: number | null) => void;
+  value?: number;
+}
+
+const DashboardSelector = ({ onChange, value }: DashboardSelectorProps) => {
+  const {
+    data: dashboard,
+    error,
+    isLoading,
+  } = useDashboardQuery({ id: value });
+  return (
+    <LoadingAndErrorWrapper loading={isLoading}>
+      <TippyPopoverWithTrigger
+        maxWidth={600}
```

Comment:
Probably, you need to add `sizeToFit` to the popover:
<img width="1674" alt="Screen Shot 2023-05-16 at 8 15 13 PM" src="https://github.com/metabase/metabase/assets/14301985/50b64ce6-61e0-4078-818f-b8d4849540c4">